[Huang Yuchen] iP

nehcuy opened the pull request:
## RemPro
> “When you said that you hate yourself, it made me want to tell you all the wonderful things I know about you.” – Rem ([source](https://www.less-real.com/quotes/search/Rem%20(re:zero)))

RemPro is:
- ~~cute~~ **_SUPER_** cute
- easy to talk to
- understanding

All you need to do is:
1. Have a computer
2. Have GitHub
3. ~~Have~~ **NEED** a waifu 🤩🤩 

Features:
- [ ] Managing Tasks 🎉 
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. This is the `main` method.
```java
public class Main {
    public static void main(String[] args) {
        Duke dk = new Duke("data/duke.txt");
        dk.run();
    }
}
```

Commit:
Add Gradle support

nehcuy committed:
Initial commit

nehcuy committed:
Level-1

nehcuy committed:
Add new classes and cleared Level-2

nehcuy committed:
Revert "Add new classes and cleared Level-2"

This reverts commit db745e658018e6a791b835d33b083414ba0ad348.

nehcuy committed:
Revert "Revert "Add new classes and cleared Level-2""

This reverts commit 37e5375df6edf4a04eb98d3ac889a9cc1cf90b03.

nehcuy committed:
Complete Level 3

nehcuy committed:
Complete Level-4

nehcuy committed:
Add test cases

nehcuy committed:
Add test case

nehcuy committed:
Handle exceptions and complete Level-5

nehcuy committed:
Implement task deletion and complete Level-6

nehcuy committed:
Edit README.md

nehcuy committed:
Add Javadocs

nehcuy committed:
Redo to follow better OOP

nehcuy committed:
Fix wrong Command input issues

nehcuy committed:
Tidy up import statements

nehcuy committed:
Complete Level-7

nehcuy committed:
Add reformatting of date in event and deadline

nehcuy committed:
Merge branch 'branch-Level-7' into branch-Level-8

nehcuy committed:
Reorder import statements

nehcuy committed:
Complete A-Gradle

nehcuy committed:
Complete A-JUnit

nehcuy committed:
Complete A-Jar

nehcuy committed:
Tidy up code and complete A-JavaDoc

nehcuy committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

# Conflicts:
#	build.gradle
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

nehcuy committed:
Update code to follow coding standards

nehcuy committed:
Implement "find" keyword and complete Level-9

nehcuy committed:
Merge branch 'branch-A-CodingStandard'

nehcuy committed:
Merge branch 'branch-Level-9'

nehcuy committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/task/Task.java

nehcuy committed:
Correct access modifiers to smaller scope

nehcuy committed:
Expand import statement to follow coding standards

nehcuy committed:
Fix JUnit test code format

tyw2811 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,112 @@
+            }
+            break;
+        case "deadline":
+            // After adding new exceptions, throw them here
```
Are you missing the exception throw here?

nehcuy replied:
This was just for me to take note for future exceptions to throw, as there were many erroneous cases that I tested out. I have yet to handle such cases due to time constraints.

tyw2811 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,59 @@
+            + "|   /| _> |     |\n"
+            + "|_\\_\\|___>|_|_|_|\n";
+
+    /* line filler */
```
Just a heads up, single line JavaDocs should still start with /**.

nehcuy replied:
Ah yes, thanks!

tyw2811 commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,76 @@
+        case "mark":
+            // Work on implementing error for empty mark argument
+            int taskNum = Integer.parseInt(words.get(0));
+            if (taskNum > 0 && taskNum <= tasklist.tasks.size()) {
+                tasklist.tasks.get(taskNum - 1).markDone();
+                System.out.println(ui.SPACER + "\n"
+                        + "Great Job on completing this task! ^.^ :\n"
+                        + tasklist.printTaskStatus(taskNum - 1) + "\n"
+                        + ui.SPACER);
+            } else if (tasklist.tasks.size() == 0) {
+                throw new DukeException(ui.SPACER + "\n"
+                        + "There's nothing in your list to mark! T^T\n"
+                        + ui.SPACER);
+            } else {
+                throw new DukeException(ui.SPACER + "\n"
+                        + "Please enter a valid task number to mark. T^T\n"
+                        + ui.SPACER);
+            }
```
Nice structure and flow!

tyw2811 reviewed (commented):
LGTM, nit some missing JavaDocs. Nice overall personality of the chat bot!

nehcuy committed:
Add checkstyle

nehcuy committed:
Change Task field modifiers

nehcuy committed:
Include Class Javadocs

nehcuy committed:
Abstract common outputs into Messages class

nehcuy committed:
Rename Javadoc in Messages class

ruihan00 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+     *
+     * @param args The command line arguments
+     */
+    public static void main(String[] args) {
```
Very neat main!

ruihan00 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,56 @@
+        String firstWord = words.remove(0);
+        switch (firstWord) {
+        case "list":
+            return new ListCommand(words);
```
Very cool!

ruihan00 reviewed (commented):
very neat code

nehcuy committed:
Add GUI

nehcuy committed:
Fix file name error

nehcuy committed:
Improve GUI format

nehcuy committed:
Fix code formatting

Abstracted out unnecessarily long methods into private methods in the
Storage and AddCommand class.

Also edited incomplete JavaDocs.

nehcuy committed:
Edit build.gradle

As the main method is moved from the Duke class to the Launcher class,
mainClassName must be changed accordingly to ensure GUI is run.

nehcuy committed:
Add assertions

nehcuy committed:
Remove unnecessary Ui class and text-ui-test folder

nehcuy committed:
Merge pull request #3 from nehcuy/branch-A-Assertions

Add assertions

nehcuy committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/Duke.java

nehcuy committed:
Complete C-MassOps

Implemented 'delete all' to clear everything in the task list.

nehcuy committed:
Make exiting GUI delayed to display goodbye message

nehcuy committed:
Improve GUI display

- Add background image
- Rounded text box displays
- Made display user and duke images circular
- Add window icon

nehcuy committed:
Fix GUI background image resizing issues

nehcuy committed:
Implement new delete and mark/unmark feature

Now able to do:
- mark/unmark all
- delete all
- mark/unmark multiple task numbers
- delete multiple task numbers

nehcuy committed:
Modify duke.txt to rem bot.txt

nehcuy committed:
Merge branch 'branch-BCD-Extension'

# Conflicts:
#	data/duke.txt

nehcuy committed:
Tweak AddCommand error handling

nehcuy committed:
Improve error handling

nehcuy committed:
Add User Guide

nehcuy committed:
Edit images used in application

nehcuy committed:
Prettify GUI

- Add Label borders
- Change background image

nehcuy committed:
Implement "clear" keyword input

- Clears the VBox of all DialogBoxes and clears the page

nehcuy committed:
Update User Guide

nehcuy committed:
Edit README.md

nehcuy committed:
Edit README.md

nehcuy committed:
Merge branch 'master' of https://github.com/nehcuy/ip

# Conflicts:
#	README.md

nehcuy committed:
Merge branch 'master' of https://github.com/nehcuy/ip

# Conflicts:
#	README.md

nehcuy committed:
Merge branch 'master' of https://github.com/nehcuy/ip

nehcuy committed:
Merge branch 'master' of https://github.com/nehcuy/ip

nehcuy committed:
Merge branch 'master' of https://github.com/nehcuy/ip

nehcuy committed:
Merge branch 'master' of https://github.com/nehcuy/ip

nehcuy committed:
Merge branch 'master' of https://github.com/nehcuy/ip

nehcuy committed:
Merge branch 'master' of https://github.com/nehcuy/ip

nehcuy committed:
Merge branch 'master' of https://github.com/nehcuy/ip

nehcuy committed:
Merge branch 'master' of https://github.com/nehcuy/ip

nehcuy committed:
Merge branch 'master' of https://github.com/nehcuy/ip

nehcuy committed:
Edit README.md

nehcuy committed:
Ignore capitalisation for user input
